ciao-controller/internal/datastore/datastore.go

```diff
@@ -1154,21 +1154,23 @@ func (ds *Datastore) HandleStats(stat payloads.Stat) error {
 		ds.addNodeStat(stat)
 	}
 
-	return ds.addInstanceStats(stat.Instances, stat.NodeUUID)
+	return errors.Wrapf(ds.addInstanceStats(stat.Instances, stat.NodeUUID), "error updating stats")
 }
 
 // HandleTraceReport stores the provided trace data in the datastore.
 func (ds *Datastore) HandleTraceReport(trace payloads.Trace) error {
+	var err error
 	for index := range trace.Frames {
 		i := trace.Frames[index]
 
-		err := ds.db.addFrameStat(i)
-		if err != nil {
-			glog.Warning(err)
+		if tmpErr := ds.db.addFrameStat(i); tmpErr != nil {
+			if err != nil {
+				err = errors.Wrapf(tmpErr, "error adding stats to database")
+			}
```

Comment:
Here I want to continue around the loop even if any one errors out. The code is not consistent on this but i'm simply trying to adhere to the previous behaviour as closely as possible. A later cleanup might decide it's fine to abort halfway through the loop.

There is a bug here though, it should say if err == nil not err != nil.